File: 198-house-robber/house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        int pre = nums[0];
        int pre2 = 0;
        for(int i =1;i<n;i++){
            int pick = nums[i];
            if(i>1){
                pick+=pre2;
            }
            int notpick = 0 + pre;
            int cur = max(pick , notpick);
            pre2 = pre;
            pre = cur;
        }
        return pre;
    }
    

};

File: 198-house-robber/house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        pre = nums[0]
        pre2 = 0
        for i in range(1,n):
            pick = nums[i]
            if(i>1):
                pick+=pre2
            notpick = 0 + pre
            cur = max(pick,notpick)
            pre2 = pre
            pre=cur
        return pre